cli/azd/pkg/apphost/aca_ingress.go

```diff
@@ -0,0 +1,172 @@
+				props.httpOnly = false
+			}
+			if binding.Transport == acaIngressTransportHttp2 {
+				props.hasHttp2 = true
+			}
+		}
+		endpointByContainerPortProperties[containerPort] = props
+	}
+
+	countExternalGroups := 0
+	for _, props := range endpointByContainerPortProperties {
+		if props.external {
+			countExternalGroups++
+		}
+		if props.external && !props.httpOnly {
+			return nil, fmt.Errorf("External non-HTTP(s) endpoints are not supported")
+		}
+	}
+	if countExternalGroups > 1 {
+		return nil, fmt.Errorf("Multiple external endpoints are not supported")
+	}
+
+	var httpOnlyGroups []string
+	for groupKey, props := range endpointByContainerPortProperties {
+		if props.httpOnly {
+			httpOnlyGroups = append(httpOnlyGroups, groupKey)
+		}
+	}
+
+	var ingress string
+	if len(httpOnlyGroups) == 1 {
+		ingress = httpOnlyGroups[0]
+	}
+
+	if ingress == "" {
+		// We have more than one, pick prefer external one
+		var externalHttpOnly []string
+		for _, groupKey := range httpOnlyGroups {
+			if endpointByContainerPortProperties[groupKey].external {
+				externalHttpOnly = append(externalHttpOnly, groupKey)
+			}
+		}
+
+		if len(externalHttpOnly) == 1 {
+			ingress = externalHttpOnly[0]
+		} else if len(httpOnlyGroups) > 1 {
+			return nil, fmt.Errorf("Multiple internal only HTTP(s) endpoints are not supported")
+		}
+	}
+
+	additionalPortsCount := len(endpointByContainerPort)
+	if ingress != "" {
+		additionalPortsCount--
+	}
+	if additionalPortsCount > 5 {
+		return nil, fmt.Errorf("More than 5 additional ports are not supported. " +
+			"See https://learn.microsoft.com/en-us/azure/container-apps/ingress-overview#tcp for more details.")
+	}
+
+	finalIngress := &genContainerAppIngress{}
+
+	if ingress != "" {
+		props := endpointByContainerPortProperties[ingress]
+
+		finalIngress.External = props.external
+		finalIngress.TargetPort = defaultIngressPort
+		finalIngress.Transport = acaIngressSchemaHttp
+		if props.hasHttp2 {
+			finalIngress.Transport = acaIngressTransportHttp2
+		}
+		if props.hasHttp2 || !props.external {
+			finalIngress.AllowInsecure = true
+		}
+
+	} else {
+		port := 0
+		for groupKey := range endpointByContainerPort {
+			port = endpointByContainerPortProperties[groupKey].port
+			ingress = groupKey
+			break
+		}
+		finalIngress.Transport = acaIngressSchemaTcp
+		finalIngress.TargetPort = port
+	}
+
+	for groupKey, props := range endpointByContainerPortProperties {
+		if groupKey == ingress {
+			continue
+		}
+		finalIngress.AdditionalPortMappings = append(finalIngress.AdditionalPortMappings,
+			genContainerAppIngressAdditionalPortMappings{
+				genContainerAppIngressPort: genContainerAppIngressPort{
+					TargetPort: props.port,
+				},
+			})
+	}
+
+	return finalIngress, nil
+}
+
+type endpointGroupProperties struct {
+	port     int
+	external bool
+	httpOnly bool
+	hasHttp2 bool
+}
+
+const (
+	acaIngressSchemaTcp      string = "tcp"
```

Comment:
Sort of a NIT, but you might consider defining three types like:

```golang
type acaIngressScheme string
type acaIngressTransport string
type acaIngressProtocol string
```

And these use these types here (and in the model) types to get a bit of type safety.  I'm just a little worried about us mixing these constants together which feels a little easier when they are all typed as string.

Reply:
I started like that, but then I need to do something like 

```go
if acaIngressScheme(string) == acaIngressSchemeHttp {
}
```

That's why I ended up just using strings. Under the assumption, as well, that this is only inside the package and not exported